add --smart-titles flag to hide_edge_borders

nim65s opened the pull request:
Hi, 

This extend the `--i3` flag to hide title bars on any kind of container.

nim65s committed:
add --smart-titles flag to hide_edge_borders

to hide the title bar on workspaces with one child, fix #7409

nim65s commented:
Well, actually, this doesn't exactly fix #7409, and only implement part of it. But I'll wait for reviews before changing things.

hashworks commented:
This is everything I expected from #7409, thanks! If someone wants to try it out on Arch Linux: Here is a [sway-git](https://aur.archlinux.org/packages/sway-git) build with this commit: https://fb.hash.works/m1ljk5bn/ (requires wlroots-git).

<details>

<summary>PKGBUILD diff</summary>

```patch
  diff --git a/PKGBUILD b/PKGBUILD
  index 77411ef..5fd1caf 100644
  --- a/PKGBUILD
  +++ b/PKGBUILD
  @@ -1,7 +1,7 @@
   # Maintainer: GreyXor <greyxor@protonmail.com>
   # Contributor: Drew DeVault <sir@cmpwn.com>
   pkgname=sway-git
  -pkgver=1.6.rc2.r584.g0aceff74
  +pkgver=1.6.rc2.r586.gfa294a90
   pkgrel=1
   pkgdesc='Tiling Wayland compositor and replacement for the i3 window manager (git development version)'
   arch=('x86_64')
  @@ -47,9 +47,11 @@ optdepends=(
   provides=("${pkgname%-git}")
   conflicts=("${pkgname%-git}")
   source=("${pkgname}::git+${url}.git"
  +"https://github.com/swaywm/sway/commit/0e37cc318e56017f881177449e8c8ecdd5641176.patch"
   "50-systemd-user.conf"
   "sway-portals.conf")
   b2sums=('SKIP'
  +        'd3a7bf9511c0b9b14e6b531b79e5d68ee3bb0dc754190fcbeedbcbcafafeee57bc187c4b1d65a789c7bd43780b01a6761fb3a8cb655cdc2b81762dea391a5271'
           '71f45f9abb4e9f98a52177b227aa30ab27d02c9eef8a31400460e71c72b6d40ec396581f0b1703d4cec655aaba704077212882f643c6efb6cda951ea69b5383d'
           'eeaa6bdfae0fa6c0445d7d02209ef9142d529f1770fd8d9d614772c276ffa7461247523399164fed70ad39b25af9a91fcf8afa23af5c193c898c44487956de7f')
   
  @@ -59,6 +61,12 @@ pkgver() {
     git describe --long --tags | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g'
   }
   
  +prepare() {
  +  cd "$pkgname"
  +
  +  git apply ../0e37cc318e56017f881177449e8c8ecdd5641176.patch
  +}
  +
   build() {
     arch-meson "$pkgname" build -D sd-bus-provider=libsystemd
     meson compile -C build
```

</details>

@nim65s I think the `--title` flag is a bit ambiguous, maybe `--lone-title` would be better?

hashworks commented:
I think I found a bug.

When one creates a single window, it works as expected, no title bar is shown:
![image](https://github.com/swaywm/sway/assets/2362091/7bb05861-e0e0-4e3a-8cd2-ad913be7792f)

Switching to vertical split mode and creating another window works fine as well, the title bars appear:
![image](https://github.com/swaywm/sway/assets/2362091/60d50c37-372d-45d0-bb7b-e746ebd1f611)

However, switching one side to horizontal mode removes the title bar on that side:
![image](https://github.com/swaywm/sway/assets/2362091/006ba93b-c797-4e88-9890-8b797f7e63dd)

When another window is created the title bars re-appear:
![image](https://github.com/swaywm/sway/assets/2362091/0f800fa8-3e2a-48e3-85d2-b0b96ff37a1f)

Removing the latter window removes the title bar again:
![image](https://github.com/swaywm/sway/assets/2362091/755c2d0c-fd40-4e1c-ac4e-7250e70545bd)

nim65s commented:
Thanks for the bug report ! I'll work on it.

Meanwhile, for the PKGBUILD, I guess it will be easier to work directly on my fork:
```diff
diff --git a/PKGBUILD b/PKGBUILD
index 77411ef..1fc73af 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -46,7 +46,7 @@ optdepends=(
 )
 provides=("${pkgname%-git}")
 conflicts=("${pkgname%-git}")
-source=("${pkgname}::git+${url}.git"
+source=("${pkgname}::git+${url/swaywm/nim65s}"
 "50-systemd-user.conf"
 "sway-portals.conf")
 b2sums=('SKIP'

nim65s commented:
@hashworks : I think I fixed the bad behaviour you described. Could you double-check it ?
Also, I agree with the naming, but I think "smart" is used in the docs and the conf, while "lone" is in the code, so I switched the option to `--smart-titles` for consistency.

hashworks commented:
Thanks, looks like that fixed it.

svalaskevicius commented:
thanks! I've cherry-picked this to my local running version after almost starting to implement the same :) 

so far this is working great!

(trying sway after years of xmonad and the titlebar has been staring at me for a few weeks now)

nim65s commented:
This can be rebased on 1.9 (ref. https://github.com/nim65s/sway/tree/1-9-smart), but not on master since #6844.
Working on this further is then currently blocked by #8072
